90zfsbootmenu/hook/zfsbootmenu-parse-commandline.sh

```diff
@@ -30,9 +33,15 @@ fi
 # Make sure a base directory exists
 mkdir -p "${BASE:=/zfsbootmenu}"
 
-if [ -z "${BYTE_ORDER}" ]; then
-  zwarn "unable to determine platform endianness; assuming little-endian"
-  BYTE_ORDER="le"
+# Only zwarn/zerror will log prior to loglevel being updated by a KCL argument
+min_logging=4
+loglevel=$( get_zbm_arg loglevel )
+if [ -n "${loglevel}" ]; then
+  # minimum log level of 4, so we never lose error or warning messages
+  [ "${loglevel}" -ge ${min_logging} ] || loglevel=${min_logging}
```

Comment:
The early source of `/lib/kmsg-logl-lib.sh` will assume the default log level and will have skipped the `traperror` stuff in that file. This isn't a huge deal because every other file should be souring the logger library, and this will redefine the functions on demand. However, we may wish to have proper logging for the rest of this file, so I recommend
```sh
if [ "${loglevel}" -ge "${min_logging}" ]; then
  source /lib/kmsg-log-lib.sh >/dev/null 2>&1
else
  loglevel="${min_logging}"
fi
```
or, if you adopt my include-guard recommendation for the logger library (see comment on kmsg-log-lib.sh), change the source line to
```sh
FORCE_RELOAD=1 source /lib/kmsg-log-lib.sh >/dev/null 2>&1
```

Reply:
Guard added, along with a re-import toggle.